x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -2745,7 +2745,7 @@ async function validateIsNotHostedPolicy(
     throw new AgentPolicyNotFoundError('Agent policy not found');
   }
 
-  if (agentPolicy.is_managed && !force) {
+  if (agentPolicy.is_managed && !agentPolicy.supports_agentless && !force) {
```

Comment:
now that has more conditions, what about moving to a variable to be clearer in the intent?

```js
const isManagedPolicyWithoutAgentlessSupport = agentPolicy.is_managed && !agentPolicy.supports_agentless && !force;

if (isManagedPolicyWithoutAgentlessSupport) {
    ...
}
```